### DISCUSSION TITLE
SOS

### DISCUSSION THREAD

**Discussion by FaysalMossi** [Keyfactor/ejbca-ce#487] [Q&A]
Hello everyone
if it's not too much to ask can you make a video on how to install EJBCA on a virtual machine from A to Z please
thanks in advance

**Comment by primetomas** [comment]
Hi Faysal,

If you are unsure how to deploy EJBCA from source. I recommend starting with the Docker container. You can easily start the container inside a VM as well.

Generic:
https://hub.docker.com/r/keyfactor/ejbca-ce

Tutorials with Videos:
https://doc.primekey.com/ejbca/tutorials-and-guides/tutorial-start-out-with-ejbca-docker-container
https://doc.primekey.com/ejbca/tutorials-and-guides/quick-start-guide-start-ejbca-container-with-unauthenticated-network-access
https://doc.primekey.com/ejbca/tutorials-and-guides/quick-start-guide-start-ejbca-container-with-client-certificate-authenticated-access

Running in Public Cloud:
https://doc.primekey.com/ejbca/tutorials-and-guides/get-started-with-ejbca-community-container-on-aws
https://doc.primekey.com/ejbca/tutorials-and-guides/get-started-with-ejbca-community-container-on-azure

**Reply by FaysalMossi** [reply]
Thank you a lot.
I tried many times to deploy it on a VM by following the documentation but in vain.

**Reply by primetomas** [reply]
You can find the documentation for that in the DockerHub documentation:
https://hub.docker.com/r/keyfactor/ejbca-ce

You customize by setting "-e" variables.

**Reply by primetomas** [reply]
See examples in the link above. If there is something specific you are looking for, just ask. Normally you don't need to customize anything/much using variables as almost everything is done in the EJBCA Application itself.

**Comment by FaysalMossi** [comment]
So  what to fix this :
![2](https://github.com/Keyfactor/ejbca-ce/assets/96942948/9cf385da-39e6-41d0-a91c-8931d7753db7)

**Reply by primetomas** [reply]
That depends on what you did. If you install with client certificate access, you need to install a _correct_ superadmin certificate in your browser. If you have had old ones on your browser, so you have multiple ones with the same name, you need to select the _right_ one when asked which certificate to use for authentication.
I recommend that you remove all old certificates from your browser and only install the superadmin.p12 that is related to your last installation.

(this answer makes some assumptions about what you did to install it, which may be incorrect)
